api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxCurrCohortQueries.java

```diff
@@ -723,4 +728,171 @@ public CohortDefinition getPatientWhoAfterMostRecentDateHaveDrusPickupOrConsulta
+    cd.addSearch("dispensation", EptsReportUtils.map(quarterlyDispensation, codedObsMappings));
+    cd.addSearch("startOrContinue", EptsReportUtils.map(startOrContinue, codedObsMappings));
+    cd.addSearch("completed", EptsReportUtils.map(completed, codedObsMappings));
+    cd.setCompositionString("(pickupDiff OR dispensation OR startOrContinue) NOT completed");
+    return cd;
+  }
+
+  public CohortDefinition getPatientsWithMoreThan6MonthsOfDispensationQuantity() {
+    CompositionCohortDefinition cd = new CompositionCohortDefinition();
+    cd.addParameter(new Parameter("onOrBefore", "Before Date", Date.class));
+    cd.addParameter(new Parameter("onOrAfter", "After Date", Date.class));
+    cd.addParameter(new Parameter("location", "Location", Location.class));
+    CohortDefinition pickupDiff = getPatientsWithNextPickupMoreThan173Days();
+    CohortDefinition quarterlyDispensation = getPatientsWithQuarterlyTypeOfDispensation();
+    CohortDefinition startOrContinue = getPatientsWithStartOrContinueOnQuarterlyDispensation();
+    CohortDefinition completed = getPatientsWithCompletedOnQuarterlyDispensation();
+    cd.addSearch("pickupDiff", mapStraightThrough(pickupDiff));
+    String codedObsMappings =
+        "onOrAfter=${onOrAfter},onOrBefore=${onOrBefore},locationList=${location}";
+    cd.addSearch("dispensation", EptsReportUtils.map(quarterlyDispensation, codedObsMappings));
+    cd.addSearch("startOrContinue", EptsReportUtils.map(startOrContinue, codedObsMappings));
+    cd.addSearch("completed", EptsReportUtils.map(completed, codedObsMappings));
+    cd.setCompositionString("(pickupDiff OR dispensation OR startOrContinue) NOT completed");
+    return cd;
+  }
+
+  private CohortDefinition getPatientsWithNextPickupMoreThan173Days() {
+    return getPatientsWithNextPickupBetweenDaysAfterLastPharmacyEncounter(173, null);
+  }
+
+  private CohortDefinition getPatientsWithNextPickup83to173Days() {
+    return getPatientsWithNextPickupBetweenDaysAfterLastPharmacyEncounter(83, 173);
+  }
+
+  private CohortDefinition getPatientsWithNextPickupLessThan83days() {
+    return getPatientsWithNextPickupBetweenDaysAfterLastPharmacyEncounter(null, 83);
+  }
+
+  private CohortDefinition getPatientsWithMonthlyTypeOfDispensation() {
+    CodedObsCohortDefinition cd = new CodedObsCohortDefinition();
+    cd.addParameter(new Parameter("onOrBefore", "Before Date", Date.class));
+    cd.addParameter(new Parameter("onOrAfter", "After Date", Date.class));
+    cd.addParameter(new Parameter("locationList", "Location", Location.class));
+    cd.addEncounterType(hivMetadata.getAdultoSeguimentoEncounterType());
+    cd.setTimeModifier(BaseObsCohortDefinition.TimeModifier.LAST);
+    cd.setQuestion(hivMetadata.getTypeOfDispensationConcept());
+    cd.setOperator(SetComparator.IN);
+    cd.addValue(hivMetadata.getMonthlyConcept());
+    return cd;
+  }
+
+  private CohortDefinition getPatientsWithQuarterlyTypeOfDispensation() {
```

Comment:
Same here. Please ask which date are comparing against? Encounter date or obs datetime. CodedCohortDefinition will always look at the obs datetime check here https://github.com/openmrs/openmrs-module-reporting/blob/master/api/src/main/java/org/openmrs/module/reporting/cohort/definition/evaluator/BaseObsCohortDefinitionEvaluator.java